### ISSUE TITLE
New linter: detect ranges that move away from the end

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2959]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

A `range` with a positive step and descending bounds, or a negative step and ascending bounds, is empty. This is easy to write accidentally and currently produces no finding.

```clojure
(range 10 0 1)
(range 0 10 -1)
```

**Describe the solution you'd like**

Add a `:range-direction` linter, enabled at warning level by default, when literal numeric bounds and step prove that the step moves away from the end value. Report the step when supplied and the end otherwise. Dynamic arguments and correctly directed ranges should not warn.

**Describe alternatives you've considered**

Runtime evaluation could reveal an empty range, but static checking gives earlier feedback and does not execute user code. Extending a general constant-expression linter would make the configuration and message less specific.

**Additional context**

I am interested in submitting a focused PR with tests and documentation if this approach is accepted.

**Comment by NoahTheDuke** [comment]
Does this happen often?

**Comment by marco-carvalho** [comment]
I did not find any occurrences in the pinned clj-kondo or Metabase regression corpora. I proposed it because the condition is deterministic and should have few false positives, but I do not have evidence that it is frequent. Based on that result, I would be comfortable making it opt-in rather than default-on, or closing it if frequency is the threshold for adding a linter.
